src/app/theme/theme.component.spec.ts

```diff
@@ -28,4 +29,109 @@ describe('ThemeComponent', () => {
   it('should be created', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should have modal-title', () => {
+    const compiled = fixture.debugElement.nativeElement;
+    expect(compiled.querySelector('h4.modal-title')).toBeTruthy();
+  });
+
+  it('should have button of class close', () => {
+    const compiled = fixture.debugElement.nativeElement;
+    expect(compiled.querySelector('button.close')).toBeTruthy();
+  });
+
+  it('should have 6 buttons of class theme-link', () => {
+    const element = fixture.debugElement.queryAll(By.css('button.theme-link'));
+    expect(element.length).toBe(6);
+  });
+
+  it('should set default theme', async(() => {
+    spyOn(component, 'setTheme').and.callThrough();
+
+    const element = fixture.debugElement.queryAll(By.css('button.theme-link'))[0];
+    const btn = element.nativeElement as HTMLElement;
+    btn.click();
+
+    fixture.detectChanges();
+    fixture.whenStable().then(() => {
+      fixture.detectChanges();
+      const theme = JSON.parse(localStorage.getItem('theme')).value;
+      expect(theme).toBe('defaultTheme');
+    });
+  }));
+
+  it('should set dark theme', async(() => {
+    spyOn(component, 'setTheme').and.callThrough();
+
+    const element = fixture.debugElement.queryAll(By.css('button.theme-link'))[1];
+    const btn = element.nativeElement as HTMLElement;
+    btn.click();
+
+    fixture.detectChanges();
+    fixture.whenStable().then(() => {
```

Comment:
Is it ok to use ```then```? Can't it be done without it?

Reply:
Actually whatever reference I got about `whenStable` with `async`, I found the same format. I don't think it'll work without this